src/__tests__/fireEvent-textInput.test.tsx

```diff
@@ -0,0 +1,131 @@
+import React from 'react';
+import { Text, TextInput, TextInputProps } from 'react-native';
+import { render, fireEvent } from '..';
+
+function WrappedTextInput(props: TextInputProps) {
+  return <TextInput {...props} />;
+}
+
+function DoubleWrappedTextInput(props: TextInputProps) {
+  return <WrappedTextInput {...props} />;
+}
```

Comment:
@mdjastrzebski i don't fully understand the use of WrappedTextInput / DoubleWrappedTextInput, could you explain it to me?

Reply:
The current `editable` check would work only on host or composite `TextInput` exported from RN:

```typescript
const isTextInput = (element?: ReactTestInstance) => {
  if (!element) {
    return false;
  }

  return (
    filterNodeByType(element, TextInput) ||
    filterNodeByType(element, getHostComponentNames().textInput)
  );
};

if (isTextInput(element)) return element?.props.editable !== false;
``` 

When user would create their own `TextInput` wrapper or use such component from some UI library, then the `editable` prop would not be taken into account, as that wrapped `TextInput` would not pass the `isTextInput` check.
That would cause all of the events on such wrapper `TextInput` to be called regardless of `editable` prop.

Reply:
It's related to host/composite components but also to nearest touch responder. @AugustinLF do you have any suggestions that would make test descriptions better?

Reply:
Added relevant comments for tests.